components/bms_worker/src/Cooling.c

```diff
@@ -0,0 +1,156 @@
+ *                               M A C R O S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                           P U B L I C  V A R S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+/**
+ * @brief  Stores the public Cooling Manager struct
+ */
+Cooling_Mngr_S COOLING;
+
+
+/******************************************************************************
+ *            P U B L I C  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *          P R I V A T E  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/**
+ * @brief  Cooling Module Init function
+ */
+static void Cooling_Init()
+{
+    COOLING.state      = COOLING_INIT;
+    COOLING.percentage = 0;
+    FANS_Init();
+
+    COOLING.state = COOLING_INIT;
+}
+
+/**
+ * @brief  Cooling Module 1Hz periodic function
+ */
+static void Cooling1Hz_PRD(void)
+{
+    static uint8_t step = 0;
+
+    switch (COOLING.state)
+    {
+        case COOLING_INIT:
+            step += 20;
+            COOLING.percentage = (step <= 100) ? step : 200 - step;
+
+            if (step >= 200)
+                COOLING.state = COOLING_OFF;
+            break;
+        case COOLING_ON:
+        case COOLING_OFF:
+            if (ENV.values.cells.max_temp < 350)
+            {
+                COOLING.percentage = 0;
+            }
+            else if (ENV.values.cells.max_temp > 550)
+            {
+                COOLING.state = COOLING_FULL;
+                COOLING.percentage = 100;
+            }
+            else if (ENV.values.cells.max_temp > 500)
+            {
+                COOLING.state = COOLING_ON;
+                COOLING.percentage = 80;
+            }
+            else if (ENV.values.cells.max_temp > 450)
+            {
+                COOLING.state = COOLING_ON;
+                COOLING.percentage = 60;
+            }
+            else if (ENV.values.cells.max_temp > 400)
+            {
+                COOLING.state = COOLING_ON;
+                COOLING.percentage = 40;
+            }
+            else if (ENV.values.cells.max_temp >= 350)
+            {
+                COOLING.state = COOLING_ON;
+                COOLING.percentage = 20;
```

Comment:
what do these states actually accomplish? seems like cooling state just represents whether percentage is 0 or not?

Reply:
Added states because in full blast theres a hysterisis to come out of it. Added ON and OFF more as information and forward compatibility if ever the cooling medium and control changes. For Cooling off can also integrate long term into low power modes if ever integrate power drawn from segments constantly